src/typing/typeloadCheck.ml

```diff
@@ -564,8 +564,10 @@ module Inheritance = struct
 					   purpose. However, we STILL have to delay the check because at the time pending is handled, the class
 					   is not built yet. See issue #10847. *)
 					pending := (fun () -> delay ctx PConnectField check_interfaces_or_delay) :: !pending
-				| _ ->
+				| _ when ctx.com.display.dms_kind = DMNone ->
```

Comment:
Checking for display modes explicitly is something we should  tone down. Ideally, we do everything through the `dms_` settings, but I'm not sure if entirely ignoring `check_interfaces` in display modes is correct in the first place.

Reply:
Would if be ok for `dms_full_typing = false`?

Reply:
Yes I think that's good.